src/compass/utils/lut.py

```diff
@@ -12,242 +12,150 @@
-
-    # Get DEM raster
+    # Dem info
     dem_raster = isce3.io.Raster(dem_path)
     epsg = dem_raster.get_epsg()
     proj = isce3.core.make_projection(epsg)
     ellipsoid = proj.ellipsoid
 
-    # Create directory to store SET temp results
+    # Get LUT spacing
+    rg_step = lut_par.range_spacing
+    az_step = lut_par.azimuth_spacing
+
+    # Create directory to temporary results
     output_path = f'{scratch_path}/corrections'
     os.makedirs(output_path, exist_ok=True)
 
-    # Compute Geometrical Steering Doppler
-    geometrical_steering_doppler = \
-        burst.doppler_induced_range_shift(range_step=rg_step, az_step=az_step)
-
-    # Compute bistatic delay
-    bistatic_delay = burst.bistatic_delay(range_step=rg_step, az_step=az_step)
-
-    # Run rdr2geo to obtain the required layers
-    # return contents: lon_path, lat_path, height_path, inc_path, head_path
-    rdr2geo_raster_paths = compute_rdr2geo_rasters(burst, dem_raster,
-                                                   output_path, rg_step,
-                                                   az_step)
-
-    # Open rdr2geo layers
-    lon, lat, height, inc_angle, head_angle = \
-        [open_raster(raster_path) for raster_path in rdr2geo_raster_paths]
-
-    # Compute azimuth FM-rate mismatch
-    az_fm_mismatch = burst.az_fm_rate_mismatch_from_llh(lat, lon, height,
+    # If any of the following corrections is enabled
+    # generate rdr2geo layers
+    rdr2geo_enabled = lut_par.azimuth_fm_rate or \
+                      lut_par.solid_earth_tides or \
+                      lut_par.ionosphere_tec or \
+                      lut_par.static_troposphere or \
+                      lut_par.weather_model_troposphere
+    if rdr2geo_enabled:
+        # return contents: lon_path, lat_path, height_path, inc_path, head_path
+        rdr2geo_raster_paths = compute_rdr2geo_rasters(burst, dem_raster,
+                                                       output_path, rg_step,
+                                                       az_step)
+        # Open rdr2geo layers
+        lon, lat, height, inc_angle, head_angle = \
+            [open_raster(raster_path) for raster_path in rdr2geo_raster_paths]
+
+    # Get the shape of the correction LUT and create empty numpy array
+    lut = burst.bistatic_delay(range_step=rg_step,
+                               az_step=az_step)
+    lut_shape = lut.data.shape
+    rg_data = np.zeros(lut_shape, dtype=np.float32)
+    az_data = np.zeros(lut_shape, dtype=np.float32)
+
+    # Initialize data list and description to save corrections
+    data_dict = {
+        'doppler': [rg_data, 'Slant range geometry and steering doppler'],
+        'bistatic_delay': [rg_data, 'Bistatic delay'],
+        'azimuth_fm_rate': [rg_data, 'Azimuth FM rate mismatch'],
+        'rg_set': [rg_data, 'Slant range Solid Earth Tides'],
+        'az_set': [rg_data, 'Azimuth Solid Earth Tides'],
+        'static_tropo': [rg_data, 'Static troposphere delay'],
```

Comment:
```suggestion
```
I think we have decided not to include static troposphere correction layer into the output. I've found in `h5_helpers.py` that you indeed skipped writing this layer.
Not sure about why you decided to put the static tropo in to `data_dict` here, but I think it would be less confusing not to include this layer at this point.

Reply:
Sounds good. Thanks for the clarification.